### DISCUSSION TITLE
Grouping improvements

### DISCUSSION THREAD

**Discussion by murphpdx** [JedWatson/react-select#5659] [Ideas]
1. Scenario: I want users to be able to choose a city or country. There are prepopulated options in the dropdown but not an exhaustive list. If the user wants to create something that's not available, we need to know if they intended to create a country or a city. The create option should show up within each grouping.  
2. Expand and collapse groupings in the popover. 
3. Options should have a label for the dropdown and for the chips. I want to display different things when the user chooses an option and it becomes a chip vs what appears in the dropdown.

Alternatively, allow us to pass in our own custom menu. Then we can customize, however.

**Comment by nick-keller** [comment]
It sounds like everything is already feasable by providing your own [components](https://react-select.com/components)
